src/services/routing.ts

```diff
@@ -102,6 +111,26 @@ export function getDrilldownSlug() {
   return slug as PageSlugs;
 }
 
+/**
+ * The "primary" label, is the replacement for the service_name paradigm
+ * It must be an indexed label with an include filter
+ * Note: Will return the label as it exists in the url, so "service_name" will be returned as "service", we'll need to adjust for this case if we want to support URLs from before this change
+ */
+export function getPrimaryLabelFromUrl(): RouteProps {
+  const location = locationService.getLocation();
+  const startOfUrl = '/a/grafana-lokiexplore-app/explore';
+  const endOfUrl = location.pathname.slice(location.pathname.indexOf(startOfUrl) + startOfUrl.length + 1);
+  const routeParams = endOfUrl.split('/');
```

Comment:
I added an error if the label name/value (formerly known as service name/value) are not in the URL, but this is another case where if the url structure was that mangled it would never route to our existing pages, but it doesn't hurt to fail loudly in case future development changes the routing.

I'm not sure about the array destructuring because the breakdownLabel is not the third element in the array but the 4th, we could define a variable that we don't do anything with but that always hurts :P

Reply:
Actually nvm I need to remove this error I just added as it gets thrown when redirecting back to the service selection when the primary label is removed.

Reply:
Fixed